### ISSUE TITLE
Re-add the -Dopenssl option to zap build

### ISSUE THREAD

**Issue by Vemahk** [zigzap/zap#62]
Hey, I'd like to bring this up for discussion at least.

I didn't even really know any of this myself until fiddling with it last night, but users of zap _could_ pass the -Dopenssl flag to the zap build from their own build.zig via the following:

```zig
const zap = b.dependency("zap", .{
    .target = target,
    .optimize = optimize,
    .openssl = true,
});
```

I find this approach much more simple than having to pass in the environment variable each time I want to build my own project (or by setting it some other way and having that be a pre-setup step to building my project).

I'd at least like the -Dopenssl flag still so I can continue to do this, but only because I find this solution rather elegant.

**Comment by renerocksai** [comment]
Thanks for bringing this up. And you are right!.

I just commented in a related issue that I had tried that approach but discarded it because I had typoed it 🤦‍♂️. (`use_openssl` instead of just `openssl`).

My solution is going to be to re-add the option, and maybe OR it with the potentially present env var.

However, it's 3am in the morning here, so I'm going to do that tomorrow if no one beats me to it with a PR 🙂.
